### PR TITLE
[Feature] Add Vault Hook

### PR DESCRIPTION
**Pull request by stuntguy3000**
**Summary**
This PR adds a Vault Hook, which adds two variables to expand uses for this API.

**Justification**
I believe the changes add more useful features and explore the realm of normal server activites (Economy)

**Use Case(s)**
- A way to provide a Hologram in-front a shop, and graphically display how much money the player has.

_Breakdown provided line by line below._

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,6 +49,11 @@
             <version>${fanciful.version}</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
+            <groupId>net.milkbowl.vault</groupId>
+            <artifactId>Vault</artifactId>
+            <version>1.2.27</version>
+        </dependency>
     </dependencies>
```
Comment (stuntguy3000) [inline pom.xml:73]: Add Vault to Maven.

File: pom.xml [hunk 2]
```diff
@@ -63,6 +68,10 @@
             <id>fanciful-repo</id>
             <url>https://raw2.github.com/mkremins/fanciful/mvn-repo/</url>
         </repository>
+        <repository>
+            <id>vault-repo</id>
+            <url>http://nexus.theyeticave.net/content/repositories/pub_releases/</url>
+        </repository>
```
Comment (stuntguy3000) [inline pom.xml:100]: Add Vault to Maven.

File: src/main/java/com/dsh105/holoapi/HoloAPI.java [hunk 3]
```diff
@@ -164,6 +167,14 @@ public void onEnable() {
...
         this.checkUpdates();
+
+        // Vault Hook
+        vaultHook = new VaultHook(this);
+
+        if (manager.getPlugin("Vault") != null && manager.getPlugin("Vault").isEnabled()) {
+            ConsoleLogger.log(Logger.LogLevel.NORMAL, "[Hook] [Vault] Detected and Hooked Vault.");
+            vaultHook.loadHook();
+        }
     }
```
Comment (stuntguy3000) [inline src/main/java/com/dsh105/holoapi/HoloAPI.java:178]: Initialise Vault Hook on startup, aswell as making sure it is enabled/there.

File: src/main/java/com/dsh105/holoapi/HoloAPI.java [hunk 4]
```diff
@@ -293,7 +304,11 @@ public boolean onCommand(CommandSender sender, Command cmd, String commandLabel,
         return false;
     }
 
+    public VaultHook getVaultHook() {
+        return vaultHook;
+    }
+
     public enum ConfigType {
-        MAIN, DATA, LANG
+        MAIN, DATA, LANG;
```
Comment (stuntguy3000) [inline src/main/java/com/dsh105/holoapi/HoloAPI.java:312]: Completly unrelated.

File: src/main/java/com/dsh105/holoapi/HoloAPI.java [hunk 5]
```diff
@@ -293,7 +304,11 @@ public boolean onCommand(CommandSender sender, Command cmd, String commandLabel,
         return false;
     }
 
+    public VaultHook getVaultHook() {
+        return vaultHook;
+    }
+
```
Comment (stuntguy3000) [inline src/main/java/com/dsh105/holoapi/HoloAPI.java:310]: Make it accessable to the world.

File: src/main/java/com/dsh105/holoapi/api/AnimatedHologram.java [hunk 6]
```diff
@@ -304,8 +304,12 @@ protected void generateAnimation(Player observer, String message, int index, dou
         horse.setY(y + diffY + 55);
         horse.setZ(z);
 
+        message = message.replace("%name%", observer.getName());
+        message = message.replace("%balance%", HoloAPI.getInstance().getVaultHook().getBalance(observer));
+        message = message.replace("%rank%", HoloAPI.getInstance().getVaultHook().getRank(observer));
+
         WrappedDataWatcher dw = new WrappedDataWatcher();
-        dw.watch(10, message.replace("%name%", observer.getName()));
+        dw.watch(10, message);
```
Comment (stuntguy3000) [inline src/main/java/com/dsh105/holoapi/api/AnimatedHologram.java:312]: Add the two new variables to the message, as well as move it separately from the one line (308) to provide room for expansion.

File: src/main/java/com/dsh105/holoapi/api/AnimatedHologram.java [hunk 7]
```diff
@@ -331,8 +335,12 @@ protected void updateNametag(Player observer, int index) {
     }
 
     protected void updateNametag(Player observer, String message, int index) {
+        message = message.replace("%name%", observer.getName());
+        message = message.replace("%balance%", HoloAPI.getInstance().getVaultHook().getBalance(observer));
+        message = message.replace("%rank%", HoloAPI.getInstance().getVaultHook().getRank(observer));
+
         WrappedDataWatcher dw = new WrappedDataWatcher();
-        dw.watch(10, message.replace("%name%", observer.getName()));
+        dw.watch(10, message);
```
Comment (stuntguy3000) [inline src/main/java/com/dsh105/holoapi/api/AnimatedHologram.java:343]: Add the two new variables to the message, as well as move it separately from the one line (308) to provide room for expansion.

File: src/main/java/com/dsh105/holoapi/api/Hologram.java [hunk 8]
```diff
@@ -453,6 +453,9 @@ protected void generate(Player observer, int index, double diffY, double x, doub
             msg = msg.replace("%time%", new SimpleDateFormat("h:mm a" + (HoloAPI.getInstance().getConfig(HoloAPI.ConfigType.MAIN).getBoolean("timezone.showZoneMarker") ? " z" : "")).format(c.getTime()));
         }
 
+        msg = msg.replace("%balance%", HoloAPI.getInstance().getVaultHook().getBalance(observer));
+        msg = msg.replace("%rank%", HoloAPI.getInstance().getVaultHook().getRank(observer));
+
```
Comment (stuntguy3000) [inline src/main/java/com/dsh105/holoapi/api/Hologram.java:458]: Add the two new variables to the message.

File: src/main/resources/plugin.yml [hunk 9]
```diff
@@ -2,6 +2,7 @@ name: ${project.name}
 version: ${project.version}
 author: [DSH105, CaptainBern]
 description: Advanced Hologram management
+softdepend: [Vault]
```
Comment (stuntguy3000) [inline src/main/resources/plugin.yml:5]: Load after Vault, but not require it.

**Comment by CaptainBern** [comment]
Pulled.
The implementation needs some work on, are you going to create the features/possible use cases you spoke of? 

EDIT:  Don't mind me, just being sleepy...

**Pull request merged**